pkg/dependencies/factory.go

```diff
@@ -770,6 +772,30 @@ func (f *Factory) WithHelm(opts ...executables.HelmOpt) *Factory {
 	return f
 }
 
+// WithHelmFactory configures the HelmFactory dependency.
+func (f *Factory) WithHelmFactory(opts ...HelmFactoryOpt) *Factory {
+	f.WithExecutableBuilder()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		if f.dependencies.HelmFactory != nil {
+			return nil
+		}
+
+		if f.registryMirror != nil {
+			opts = append(opts, WithRegistryMirror(f.registryMirror))
```

Comment:
It seems to me than in the CLI you don't need to build the client in runtime, so you don't really need a client. I think here is where we make use of dependency injection and inject a different `HelmClientFactory` implementation. In this case, your factory can just a dumb pass through that always returns the same client.

That way you keep the templater code common to both cli and controller and you don't pollute the factory interface with options that are only required in one case and don't make much sense in the client factory abstraction.

Reply:
Okay got it. I like that better.

Though, I think in the different `HelmClientFactory` implementation will have to know how to log in as well (since we're also we moving the registry login call to the factory). So, CLI would still be reading the environment for registry mirror details.

Reply:
> Okay got it. I like that better.
> 
> Though, it seems to me that `HelmClientFactory` implementation will have to know how to log in as well (since we're also we moving the registry login call to the factory). So, CLI would still be reading the environment for registry mirror details.